main.cc

```diff
@@ -98,10 +100,10 @@ class Matrix_t : public repl_new
 private:
     Matrix_t(Index_t nrows) : nrows_(nrows)
     {
-        nrows_per_nodelet_ = nrows_ + nrows_ % NODELETS();
-
+        nrows_per_nodelet_ = r_map(nrows_) + n_map(nrows_); 
         rows_ = (ppRow_t)mw_malloc2d(NODELETS(),
                                      nrows_per_nodelet_ * sizeof(Row_t));
+	// printf("\n %ld, %ld ", nrows_per_nodelet_, nrows_);
```

Comment:
Don't leave commented out code in your commits.

Reply:
Fixed in the next commit